lib/ladybug/energy_model/space_type.rb

```diff
@@ -0,0 +1,284 @@
+           openstudio_infiltration.setTemperatureTermCoefficient(@@schema[:definitions][:InfiltrationAbridged][:properties][:temperature_coefficient][:default])
+         end
+         if @hash[:infiltration][:velocity_coefficient]
+           openstudio_infiltration.setVelocityTermCoefficient(@hash[:infiltration][:velocity_coefficient])
+         else 
+           openstudio_infiltration.setVelocityTermCoefficient(@@schema[:definitions][:InfiltrationAbridged][:properties][:velocity_coefficient][:default])
+         end
+         infiltration_schedule_object = nil
+         infiltration_schedule = openstudio_model.getScheduleByName(@hash[:infiltration][:schedule])
+         unless infiltration_schedule.empty?
+           infiltration_schedule_object = infiltration_schedule.get
+         end
+         openstudio_infiltration.setSchedule(infiltration_schedule_object)
+        end
+      
+        if @hash[:ventilation]
+         openstudio_ventilation = OpenStudio::Model::DesignSpecificationOutdoorAir.new(openstudio_model)
+         openstudio_ventilation.setName(@hash[:ventilation][:name])
+         if @hash[:ventilation][:air_changes_per_hour]
+           openstudio_ventilation.setOutdoorAirFlowAirChangesperHour(@hash[:ventilation][:air_changes_per_hour])
+         else
+           openstudio_ventilation.setOutdoorAirFlowAirChangesperHour(@@schema[:definitions][:VentilationAbridged][:properties][:air_changes_per_hour][:default])
+         end
+         #TODO if @hash[:ventilation][:flow_per_zone]
+         if @hash[:ventilation][:flow_per_person]
+           openstudio_ventilation.setOutdoorAirFlowperPerson(@hash[:ventilation][:flow_per_person])
+         else 
+           openstudio_ventilation.setOutdoorAirFlowperPerson(@@schema[:definitions][:VentilationAbridged][:properties][:flow_per_person][:default])
+         end
+         if @hash[:ventilation][:flow_per_area]
+           openstudio_ventilation.setOutdoorAirFlowperFloorArea(@hash[:ventilation][:flow_per_area])
+         else 
+           openstudio_ventilation.setOutdoorAirFlowperPerson(@@schema[:definitions][:VentilationAbridged][:properties][:flow_per_area][:default])
+         end
+        end
+
+        openstudio_thermostat = OpenStudio::Model::ThermostatSetpointDualSetpoint.new(openstudio_model)
+        heating_schedule_object = nil
+        heating_schedule = openstudio_model.getScheduleByName(@hash[:setpoint][:heating_schedule])
+        unless heating_schedule.empty?
+          heating_schedule_object = heating_schedule.get
+        end
+        openstudio_thermostat.setHeatingSetpointTemperatureSchedule(heating_schedule_object)
+        
+        cooling_schedule_object = nil
+        cooling_schedule = openstudio_model.getScheduleByName(@hash[:setpoint][:cooling_schedule])
+        unless cooling_schedule.empty?
+          cooling_schedule_object = cooling_schedule.get
+        end
+        openstudio_thermostat.setCoolingSetpointTemperatureSchedule(cooling_schedule_object)
```

Comment:
Yea, you definitely want this to be it's own module. You want to create the thermostat when you have a zone to apply it to. Doing it for the whole space type is not going to make it applicable to all of the Zones.